sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/Metrics.java

```diff
@@ -0,0 +1,312 @@
+     * @return The created tagged timer instance.
+     */
+    public static Timer timer(String metricName, String tagKey, String tagValue) {
+        return timer(metricName, Tags.of(tagKey, tagValue));
+    }
+
+    /**
+     * Creates a timer with a set of tags to measure the duration of an operation with specific tags.
+     *
+     * @param metricName The name of the metric.
+     * @param tags The set of tags.
+     * @return The created timer instance with tags.
+     */
+    public static Timer timer(String metricName, Tags tags) {
+        return timer(metricName, tags, null);
+    }
+
+    /**
+     * Creates a timer with a set of tags and a description to measure the duration of an operation.
+     *
+     * @param metricName The name of the metric.
+     * @param tags The set of tags.
+     * @param description The description of the timer.
+     * @return The created timer instance with tags and description.
+     */
+    public static Timer timer(String metricName, Tags tags, String description) {
+        return metric.timer(metricName, tags, description);
+    }
+
+    /**
+     * Creates a Summary metric without any tags.
+     *
+     * @param metricName the name of the metric
+     * @return the created Summary metric
+     */
+    public static Summary summary(String metricName) {
+        return summary(metricName, null);
+    }
+
+    /**
+     * Creates a Summary metric and adds a single tag to it.
+     *
+     * @param metricName the name of the metric
+     * @param tagKey the key of the tag
+     * @param tagValue the value of the tag
+     * @return the created Summary metric
+     */
+    public static Summary summary(String metricName, String tagKey, String tagValue) {
+        return summary(metricName, Tags.of(tagKey, tagValue));
+    }
+
+    /**
+     * Creates a Summary metric and adds a set of tags to it.
+     *
+     * @param metricName the name of the metric
+     * @param tags the set of tags
+     * @return the created Summary metric
+     */
+    public static Summary summary(String metricName, Tags tags) {
+        return summary(metricName, tags, null, null);
+    }
+
+    /**
+     * Creates a highly customizable Summary metric that can include tags, a description, and distribution statistics
+     * configuration.
+     *
+     * @param metricName the name of the metric
+     * @param tags the set of tags
+     * @param description the description of the metric
+     * @param distributionStatisticConfig the distribution statistics configuration
+     * @return the created Summary metric
+     */
+    public static Summary summary(String metricName, Tags tags, String description,
+            DistributionStatisticConfig distributionStatisticConfig) {
+        return metric.summary(metricName, tags, description, distributionStatisticConfig);
+    }
+
+    /**
+     * NoopMetric
+     *
+     * @since 2024-08-17
+     */
+    static class NoopMetric implements Metric {
```

Comment:
why called NoopMetric

Reply:
Set the default implementation, the real implementation is in agentcore-implement